source/cl/source/device.cpp

```diff
@@ -330,6 +331,25 @@ _cl_device_id::_cl_device_id(cl_platform_id platform,
 #endif
 }
 
+void _cl_device_id::RegisterCommandQueue(cl_command_queue queue) {
+  registered_queues.push_back(queue);
+}
+void _cl_device_id::DeregisterCommandQueue(cl_command_queue queue) {
+  registered_queues.remove_if(
+      [queue](cl_command_queue q) { return q == queue; });
```

Comment:
`remove_if` is O(n), is it possible to use a different data structure instead? Maybe `std::set`?